Std/CodeAction/Basic.lean

```diff
@@ -9,6 +9,31 @@ import Std.Lean.Name
 import Std.Lean.InfoTree
 import Std.CodeAction.Attr
 
+/-!
+# Helper constructors for WorkspaceEdit with document version
+
+In core lean, the version number is always 0.
+See Lean.Lsp.WorkspaceEdit.ofTextEdit in Lean.Data.Lsp.Basic.
+-/
+
+namespace Lean.Lsp.WorkspaceEdit
+
+/-- Construct a WorkspaceEdit from a VersionedTextDocumentIdentifier and a TextEdit --/
+def ofVersionedTextEdit (td : VersionedTextDocumentIdentifier) (te : TextEdit) : WorkspaceEdit :=
+  ofTextDocumentEdit { textDocument := td, edits := #[te]}
+
+end Lean.Lsp.WorkspaceEdit
+
+namespace Lean.Server.FileWorker.EditableDocument
+
+/-- Construct a VersionedTextDocumentIdentifier from an EditableDocument --/
+def textDocument (ed : EditableDocument) : Lsp.VersionedTextDocumentIdentifier := {
```

Comment:
`textDocument` is the usual name in the LSP spec for JSON fields of this type (and the related types TextDocumentIdentifier and OptionalVersionedTextDocumentIdentifier), and also in Lean. For example in `Lean.Data.Lsp.TextSync.lean`,
```
structure DidChangeTextDocumentParams where
  textDocument : VersionedTextDocumentIdentifier
  contentChanges : Array TextDocumentContentChangeEvent
  deriving ToJson, FromJson
```

Reply:
LSP uses `textDocument` to refer to both `TextDocumentItem`s and various kinds of identifiers, and the Lean LSP data structures mirror that convention so that we can derive the corresponding JSON handlers. Regardless, I don't think that confusing the two is something that we should do in our own code, especially in server code that usually works with an `EditableDocument` and refers to it as `doc`.

Reply:
True, and my initial instinct was to use `.versionedTextDocumentIdentifier`; I'd be happy with that, but if it needs an abbreviation, `textDocument` seems natural. It needn't be construed as confusion.
Just my opinion. I won't quibble if you make a decision.

Reply:
I would prefer `versionedIdentifier`, as the code using it will typically read as `doc.versionedIdentifier`.

Reply:
Renamed to `.versionedIdentifier`.